server/controllers/student/deleteStudent.js

```diff
@@ -0,0 +1,32 @@
+const { updateStudentStatus, deleteStudent } = require('../../database/queries/deleteStudent');
+
+exports.deleteStudent = (req, res) => {
+  const { id } = req.params;
+  updateStudentStatus(id)
+    .then((student) => {
+      if (!student.rowCount) {
+        return res.status(404).send({
+          error: 'student not found',
+          data: null,
+        });
+      } return deleteStudent(id);
+    }).then((studentInfo) => {
+      if (!studentInfo.rowCount) {
```

Comment:
you don't need to check the row count again because you already did it in previous query

Reply:
This check for the second query because we update the status of is_active and then delete all applied course